semeio/workflows/ahm_analysis/ahmanalysis.py

```diff
@@ -0,0 +1,456 @@
+    run_context = ErtRunContext.ensemble_smoother_update(
+        ert.getEnkfFsManager().getFileSystem(prior_name),
+        ert.getEnkfFsManager().getFileSystem(target_name),
+    )
+    ESUpdate(ert).smootherUpdate(run_context)
+
+
+def _get_field_params(ert, ensemble_size, field_parameters, target_name):
+    """
+    Because the FIELD parameters are not exposed in the Python API we have to
+    export them to file and read them back again. When they are exposed in the API
+    this function should be updated.
+    """
+    field_data = {}
+    file_system = ert.getEnkfFsManager().getFileSystem(target_name)
+    with tempfile.TemporaryDirectory() as fout:
+        for field_param in field_parameters:
+            config_node = ert.ensembleConfig()[field_param]
+            ext = config_node.get_enkf_outfile().rsplit(".")[-1]
+            file_path = os.path.join(fout, "%d_" + field_param + "." + ext)
+            _export_field_param(config_node, file_system, ensemble_size, file_path)
+            fnames = glob.glob(os.path.join(fout, "*" + field_param + "*"))
+            field_data[field_param] = _import_field_param(
+                ert.eclConfig().get_gridfile(), field_param, fnames
+            )
+    return field_data
+
+
+def _import_field_param(input_grid, param_name, files):
+    mygrid_char = Grid(input_grid.rsplit(".", 1)[0], fformat="eclipserun")
+    all_input = []
+    for file_path in files:
+        proproff = GridProperty(file_path, name=param_name, grid=mygrid_char)
+        array_nb = proproff.get_npvalues1d(activeonly=False, fill_value=0, order="C")
+        all_input.append(array_nb)
+    return all_input
+
+
+def _export_field_param(config_node, file_system, ensemble_size, output_path):
+    # Get/export the updated Field parameters
+    EnkfNode.exportMany(
+        config_node,
+        output_path,
+        file_system,
+        np.arange(0, ensemble_size),
+    )
+
+
+def initialize_emptydf():
+    ks_data = pd.DataFrame()
+    active_obs = pd.DataFrame()
+    misfitval = pd.DataFrame()
+    return ks_data, active_obs, misfitval
+
+
+def make_update_log_df(update_log_file, key):
+    """ Read update_log file to get active and inactive observations """
+    list_of_files = [
+        os.path.join(update_log_file, f) for f in os.listdir(update_log_file)
+    ]
+    if len(list_of_files) > 1:
+        raise OSError("ERROR more than one update_log_file.")
+    # read file
+    updatelog = pd.read_csv(
+        list_of_files[0],
+        delim_whitespace=True,
+        skiprows=6,
+        usecols=[0, 2, 3, 5, 6, 8, 10],
+        header=None,
+    )
+    # define header
+    updatelog.columns = [
+        "obs_number",
+        "obs_key",
+        "obs_mean",
+        "obs_std",
+        "status",
+        "sim_mean",
+        "sim_std",
+    ]
+    # drop last line ('=====...====')
+    # drop last row i.e. end of file line
+    updatelog.drop(updatelog.tail(1).index, inplace=True)
```

Comment:
We are already skipping rows, you can also drop last line by using `skipfooter`

Reply:
ok, done